### PR TITLE
Add 'filename' attribute to traceSignals

### PR DESCRIPTION
**Pull request by punkkeks**
Additional to the directory attribute, I'd like to have a filename attribute. The reason is, when I run tests, the resulting vcd files are all named the same, that will result in the unreadable date suffix. 
The reason for not changing the name attribute is, that myhdl uses it to extract the hierachy and will rename it in the vcd file, which is ugly.

### CODE REVIEW

**Comment by cfelton** [comment]
@punkkeks The `.name` attribute is used for this.  It is the same in `traceSignals`, `toVerilog`, and `toVHDL`.

File: myhdl/_traceSignals.py [hunk 1]
```diff
@@ -89,8 +91,13 @@ def __call__(self, dut, *args, **kwargs):
             else:
                 directory = self.directory
 
+            if self.filename is None:
+                filename = self.name
+            else
```
Comment (cfelton) [inline myhdl/_traceSignals.py]: missing the colon, I don't understand the description why the `name` attribute can't be use.

**Comment by punkkeks** [comment]
No, name and filename is different, because, as I said, myhdl uses name to extract and create the hierachy in the vcd file. An example (the upper row is made with name, the lower one with filename)
screenshot from 2016-01-30 21-34-34

See https://github.com/jandecaluwe/myhdl/blob/master/myhdl/_traceSignals.py#L92

**Comment by nicmcd** [comment]
I agree with @punkkeks. I've been using the `name` attribute to adjust the filename, but what I really want is to adjust the filename, not the top level instance name within the VCD file.

**Comment by nicmcd** [comment]
@punkkeks, I think you should modify `myhdl/test/core/test_traceSignals.py` to reflect the desired changes.

**Comment by cfelton** [comment]
I understand and agree (now) but for me I never noticed or thought it was an issue that the top-level name in the VCD hierarchy changed (match the filename).  This change will affect a lot of existing code. 

I wonder if the `.name` attribute will be used if there is a `filename` attribute?

**Comment by punkkeks** [comment]
@cfelton This change may affect some code, but won't change any behavoir, because when you don't set it, it will use the original name (have you seen the diff already?!)
@nicmcd I don't know :/ As far as I can see, the test is really not a good one (at least that one for setdirectory), because it relys on internal behavoir (top.__name__ and fun.__name__) and when you change th

**Comment by punkkeks** [comment]
Darn you touchpad! I wasn't ment to close this PR.

**Comment by cfelton** [comment]
@punkkeks ok, that make sense and sounds reasonable.

**Comment by cfelton** [comment]
@punkkeks the code looks incomplete, I see where `filename` will default to `name` if it is not set but then `filename` is not used.  The `vcdpath` is still the combination of `directory` and `name`.

**Comment by punkkeks** [comment]
D'oh! Thanks for the hint. I was comiiting before thinking... In the first commit it was right. (4 commits, for such simple fix? I should go to sleep :/ )

**Comment by jandecaluwe** [comment]
Thanks.

**Pull request merged**